### ISSUE TITLE
Multiple 'len' variable in afl-mutations.h

### ISSUE THREAD

**Issue by futhewo** [AFLplusplus/AFLplusplus#1938]
Hello everyone,

Sorry for all the bugs I post.

In afl_mutate, in the MUT_INSERTASCIINUM case, there are two 'len' variables, one that comes as function arg, the other that is created inside the case:
```
case MUT_INSERTASCIINUM: {

  u32 len = 1 + rand_below(afl, 8);
  u32 pos = rand_below(afl, len);
```
They should be different: one for the global testcase length (len) and one for the asciinum length (numlen below).

I think the solution is probably as follows:
```
case MUT_INSERTASCIINUM: {

  u32 numlen = 1 + rand_below(afl, 8);
  u32 pos = rand_below(afl, len);

  /* Insert ascii number. */
  if (unlikely(len < pos + numlen)) {

    if (unlikely(len < 8)) {

      break;

    } else {

      goto retry_havoc_step;

    }

  }

  u64  val = rand_next(afl);
  char numbuf[32];
  snprintf(numbuf, sizeof(numbuf), "%llu", val);
  memcpy(buf + pos, numbuf, numlen);

  break;

}
```

Though, correct me if I am wrong, but for me, it will be a bit more efficient like so:
```
case MUT_INSERTASCIINUM: {

  if (unlikely(len < 1)) {
    break;
  }
	
  u32 numlen = 1 + rand_below(afl, min(len, 8));
  u32 pos = rand_below(afl, len - numlen);

  /* Insert ascii number. */
  u64  val = rand_next(afl);
  char numbuf[32];
  snprintf(numbuf, sizeof(numbuf), "%08llu", val);
  memcpy(buf + pos, numbuf, numlen);

  break;

}
```

The "%08llu" prevents a very rare bug, e.g. if val is '3', it will copy "00000003" instead of what remains in numbuf.

If I understand this function clearly, it does not insert an ascii num value, but rather writes over the testcase.
It is just a remark, as both overwriting and inserting are fine. I was just puzzled by the name.

Best regards

**Comment by domenukk** [comment]
@andreafioraldi can you take a look?

**Comment by vanhauser-thc** [comment]
@domenukk why do you assign andrea? it is code that I wrote.

@futhewo thank you! fixed in 7fabe5052bd41deec72fad43acd5219b5f506ac0

**Comment by futhewo** [comment]
Nice, thanks for the patch.
